aspnetcore/mvc/models/validation.md

```diff
@@ -232,69 +286,110 @@ $.get({
 
-Now when users enter an email, JavaScript in the view makes a remote call to see if that email has been taken and, if so, displays the error message. Otherwise, the user can submit the form as usual.
+* In your custom validation attribute, implement the `IClientModelValidator` interface and create an `AddValidation` method. In the `AddValidation` method, add `data-` attributes for validation, as shown in the following example:
 
-The `AdditionalFields` property of the `[Remote]` attribute is useful for validating combinations of fields against data on the server. For example, if the `User` model from above had two additional properties called `FirstName` and `LastName`, you might want to verify that no existing users already have that pair of names. You define the new properties as shown in the following code:
+  [!code-csharp[](validation/sample/Attributes/ClassicMovie2Attribute.cs?name=snippet_ClassicMovie2Attribute)]
 
-[!code-csharp[](validation/sample/User.cs?name=snippet_UserNameProperties)]
+## [Remote] attribute
 
-`AdditionalFields` could've been set explicitly to the strings `"FirstName"` and `"LastName"`, but using the [`nameof`](/dotnet/csharp/language-reference/keywords/nameof) operator like this simplifies later refactoring. The action method to perform the validation must then accept two arguments, one for the value of `FirstName` and one for the value of `LastName`.
+The `[Remote]` attribute implements client-side validation that requires calling a method on the server to determine whether field input is valid. For example, the app may need to verify whether a user name is already in use.
 
-[!code-csharp[](validation/sample/UsersController.cs?name=snippet_VerifyName)]
+To implement remote validation:
 
-Now when users enter a first and last name, JavaScript:
+1. Create an action method for JavaScript to call.  The jQuery Validate [remote](https://jqueryvalidation.org/remote-method/) method expects a JSON response:
 
-* Makes a remote call to see if that pair of names has been taken.
-* If the pair has been taken, an error message is displayed.
-* If not taken, the user can submit the form.
+   * `"true"` means the input data is valid.
+   * `"false"`, `undefined`, or `null` means the input is invalid.  Display the default error message.
+   * Any other string means the input is invalid. Display the string as a custom error message.
 
-If you need to validate two or more additional fields with the `[Remote]` attribute, you provide them as a comma-delimited list. For example, to add a `MiddleName` property to the model, set the `[Remote]` attribute as shown in the following code:
+   Here's an example of an action method that returns a custom error message:
+
+   [!code-csharp[](validation/sample/Controllers/UsersController.cs?name=snippet_VerifyEmail)]
+
+1. In your model class, annotate the property with a `[Remote]` attribute that points to the validation action method, as shown in the following example:
```

Comment:
1. In the model class, annotate the property with a `[Remote]` attribute that points to the validation action method, as shown in the following example: